v2/cmd/csv.go

```diff
@@ -0,0 +1,281 @@
+				}
+				licensePath := info.licensePath
+				if info.subModulePath != "" {
+					licensePath = info.subModulePath + "/" + info.licensePath
+				}
+				url, err = repo.RemoteUrl(ghutils.RemoteUrlArgs{
+					Path:      licensePath,
+					Version:   goModule.Version,
+					LineStart: info.lineStart,
+					LineEnd:   info.lineEnd,
+				})
+				if err != nil {
+					return err
+				}
+			}
+			moduleString := goModule.ImportPath
+			if info.subModulePath != "" {
+				moduleString = moduleString + "/" + info.subModulePath
+			}
+			_, err := f.WriteString(fmt.Sprintf(
+				"%s, %s, %s\n",
+				moduleString,
+				url,
+				info.spdxId))
+			if err != nil {
+				return fmt.Errorf("Failed to write string: %w", err)
+			}
+			licenseCount = licenseCount + 1
+			return nil
+		}
+
+		if len(override.License.Path) > 0 {
+			license := override.License
+			if len(license.SpdxId) == 0 {
+				report(fmt.Errorf("override.license.spdxId required"))
+				continue
+			}
+			klog.V(4).InfoS("License overridden", "module", goModule.ImportPath, "version", goModule.Version, "srcDir", goModule.SrcDir)
+			klog.V(5).InfoS("Override config", "override", fmt.Sprintf("%+v", override))
+			err := writeLicenseInfo(licenseInfo{
+				url:         license.Url,
+				licensePath: license.Path,
+				spdxId:      license.SpdxId,
+				lineStart:   license.LineStart,
+				lineEnd:     license.LineEnd,
+			})
+			if err != nil {
+				return err
+			}
+			for _, subModule := range override.SubModules {
+				license := subModule.License
+				if len(subModule.Path) == 0 || len(license.Path) == 0 || len(license.SpdxId) == 0 {
+					report(fmt.Errorf("override.subModule: path, license.path and license.spdxId are required: subModule=%+v", subModule))
+					continue
+				}
+				err := writeLicenseInfo(licenseInfo{
+					url:           license.Url,
+					licensePath:   license.Path,
+					spdxId:        license.SpdxId,
+					lineStart:     license.LineStart,
+					lineEnd:       license.LineEnd,
+					subModulePath: subModule.Path,
+				})
+				if err != nil {
+					return err
+				}
+			}
+			continue
+		}
+
+		klog.V(4).InfoS("Scanning", "module", goModule.ImportPath, "version", goModule.Version, "srcDir", goModule.SrcDir)
+		licensesFound, err := licenses.ScanDir(goModule.SrcDir, licenses.ScanDirOptions{ExcludePaths: override.ExcludePaths, DbPath: config.Module.LicenseDB.Path})
+		if err != nil {
+			report(err)
+			continue
+		}
+		if len(licensesFound) == 0 {
+			report(errors.Errorf("licenses not found"))
+			continue
+		}
+
+		for _, license := range licensesFound {
+			klog.V(3).InfoS("License", "module", goModule.ImportPath, "SpdxId", license.SpdxId, "path", filepath.Join(goModule.SrcDir, license.Path))
+			writeLicenseInfo(licenseInfo{
+				spdxId:      license.SpdxId,
+				licensePath: license.Path,
+			})
+			if err != nil {
+				return err
+			}
+		}
+	}
+	if errorCount > 0 {
+		return fmt.Errorf("Failed to scan licenses for %v module(s)", errorCount)
```

Comment:
Note that the default logging level is warning and users will easily see the entire list of errors because I added klog logging as I encounter each error (and collect them at the same time for a summary at the end).

When I examine licenses, the partially written file is super useful, because I don't need to wait the entire command to finish.

Can you elaborate more why the partially written file can be harmful?